tests/std/tests/GH_005421_vector_algorithms_integer_class_type_iterator/test.cpp

```diff
@@ -0,0 +1,320 @@
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <__msvc_int128.hpp>
+#include <algorithm>
+#include <cassert>
+#include <compare>
+#include <iterator>
+
+using namespace std;
+
+enum class magic_word : unsigned short {}; // prevent vectorization paths from construction iterator from pointer
+
+template <class UnderlyingPtr>
+class angry_contiguous_iterator {
+public:
+    using iterator_category    = contiguous_iterator_tag;
+    using reference            = typename iterator_traits<UnderlyingPtr>::reference;
+    using value_type           = typename iterator_traits<UnderlyingPtr>::value_type;
+    using pointer              = typename iterator_traits<UnderlyingPtr>::pointer;
+    using underlying_diff_type = typename iterator_traits<UnderlyingPtr>::difference_type;
+    using difference_type      = _Signed128;
+
+    angry_contiguous_iterator() : ptr{} {}
+    angry_contiguous_iterator(const UnderlyingPtr ptr_, magic_word) : ptr(ptr_) {}
+
+    reference operator*() const {
+        return *ptr;
+    }
+
+    pointer operator->() const {
+        return ptr;
+    }
+
+    decltype(auto) operator++() {
+        ++ptr;
+        return *this;
+    }
+
+    auto operator++(int) {
+        angry_contiguous_iterator old(*this);
+        ++ptr;
+        return old;
+    }
+
+    decltype(auto) operator--() {
+        --ptr;
+        return *this;
+    }
+
+    auto operator--(int) {
+        angry_contiguous_iterator old(*this);
+        --ptr;
+        return old;
+    }
+
+    auto operator<=>(const angry_contiguous_iterator o) const {
+        return ptr <=> o.ptr;
+    }
+
+    auto operator==(const angry_contiguous_iterator o) const {
+        return ptr == o.ptr;
+    }
+
+    friend auto operator+(const difference_type diff, const angry_contiguous_iterator it) {
+        return angry_contiguous_iterator{static_cast<underlying_diff_type>(diff) + it.ptr, magic_word{}};
+    }
+
+    friend auto operator+(const angry_contiguous_iterator it, const difference_type diff) {
+        return angry_contiguous_iterator{it.ptr + static_cast<underlying_diff_type>(diff), magic_word{}};
+    }
+
+    friend auto operator-(const angry_contiguous_iterator it, const difference_type diff) {
+        return angry_contiguous_iterator{it.ptr - static_cast<underlying_diff_type>(diff), magic_word{}};
+    }
+
+    friend auto operator-(const angry_contiguous_iterator it, angry_contiguous_iterator it_other) {
+        return difference_type{it.ptr - it_other.ptr};
+    }
+
+    decltype(auto) operator+=(const difference_type diff) {
+        ptr += static_cast<underlying_diff_type>(diff);
+        return *this;
+    }
+
+    decltype(auto) operator-=(const difference_type diff) {
+        ptr -= static_cast<underlying_diff_type>(diff);
+        return *this;
+    }
+
+    decltype(auto) operator[](const difference_type diff) const {
+        return ptr[static_cast<underlying_diff_type>(diff)];
+    }
+
+    template <class T>

```

Comment:
Should I use abbreviated templates here? It is the "type is so unimportant, doesn't even deserve a name" case.

Reply:
There is possibly a blocking issue, see DevCom-10735214. But I wonder whether we can reuse `test::redifference_iterator`.

Reply:
Oh, I found there's difference range, but didn't realize a single iterator from it can also be reused. Will try.

Reply:
Yes. This way we lose protection from accidental from pointer construction, but I think it is not that important.